R/utils.R

```diff
@@ -383,16 +383,17 @@ matching_variables <- function(variable_filters, variables) {
   not_found <- NULL
   variable_filters <- unrepair_variable_names(variable_filters)
   variables <- unrepair_variable_names(variables)
-  selected <- rep(FALSE, length(variables))
+  selected <- c()
   for (p in variable_filters) {
-      matches <- variables == p | startsWith(variables, paste0(p, "."))
-      if (!any(matches)) {
-        not_found <- c(not_found, p)
-      }
-      selected <- selected | matches
+    matches <- which(variables == p | startsWith(variables, paste0(p, ".")))
```

Comment:
if we use `which` we get the position (index) of the match, so this preserves the order